src/features/core/deviceModeTransformation/dmtHandler.js

```diff
@@ -0,0 +1,190 @@
+                });
+
+                resolve({
+                  transformedPayload: transformationResponse,
+                  transformationServerAccess: true,
+                });
+                return;
+              }
+              if (status === 400) {
+                const errorMessage = response
+                  ? `[Transformation]:: ${response}`
+                  : `[Transformation]:: Invalid request payload`;
+                reject(errorMessage);
+                return;
+              }
+              if (status === 404) {
+                resolve({
+                  transformedPayload: payload.batch,
+                  transformationServerAccess: false,
+                });
+                return;
+              }
+
+              // If the request is not successful
+              // one or more transformation is unsuccessfull
+              // retry till the retryAttempt is exhausted
+              if (retryAttempt > 0) {
+                const newretryAttempt = retryAttempt - 1;
+                setTimeout(() => this.sendEventForTransformation(payload, newretryAttempt)
+                    .then(resolve)
+                    .catch(reject), 500);
+              } else {
+                // Even after all the retries event transformation
+                // is not successful, ignore the event
+                reject(`[Transformation]:: Transformation failed with status ${status}`);
+                return;
+              }
+            } catch (err) {
+              reject(err);
+            }
+          }
+        };
+        xhr.send(JSON.stringify(payload, replacer));
+      } catch (error) {
+        reject(error);
+      }
+    });
+  }
+
+  checkQueueLengthAndProcess() {
+    if (this.queue.length > 0) {
+      this.process();
+    }
+  }
+
+  /**
+   * A helper function that will process the transformation
+   * and return the transformed event payload
+   *
+   */
+  process() {
+    this.isTransformationProcessing = true;
+    const firstElement = this.queue.shift();
+    const payload = createPayload(firstElement.event, this.authToken);
+
+    // Send event for transformation with payload, writekey and retryAttempt
+    this.sendEventForTransformation(payload, this.retryAttempt)
+      .then((outcome) => {
+        this.isTransformationProcessing = false;
+        firstElement.cb(outcome);
+        this.checkQueueLengthAndProcess();
+      })
+      .catch((err) => {
+        if (typeof err === 'string') {
+          logger.error(err);
+        } else {
+          logger.error(err.message);
+        }
+        this.isTransformationProcessing = false;
+        // send null as response in case of error or retry fail
+        firstElement.cb({ transformedPayload: null });
+        this.checkQueueLengthAndProcess();
+      });
+  }
+
+  start() {
+    const self = this;
+    setInterval(() => {
+      if (!self.isTransformationProcessing && self.queue.length > 0) {
+        self.process();
+      }
+    }, 100);
```

Comment:
can we make all the hardcoded time values as constants in the top of the file please?

Reply:
Will address this